File: pinyougouparent/pinyougou-cart-web/src/main/webapp/js/service/cartService.js
app.service('cartService',function($http){
    //购物车列表
    this.findCartList=function(){
        return $http.get('/cart/findCartList.do');
    }
    //向购物车添加商品
    this.addGoodsToCartList=function (num,itemId) {
        return $http.get('/cart/addGoodsToCartList.do?num='+num+"&itemId="+itemId);
    }

    this.findAdressList=function () {
        return $http.get('/address/findAdressList.do')
    }
    
    this.submitOrder=function (order) {
        return $http.post('/order/add.do',order);
        
    }
});

File: pinyougouparent/pinyougou-cart-interface/src/main/java/com/pinyougou/cart/service/CartService.java
package com.pinyougou.cart.service;

import com.pinyougou.pojogroup.Cart;

import java.util.List;

/**
 * @author 三国的包子
 * @version 1.0
 * @description 描述
 * @title 标题
 * @package com.pinyougou.cart.service
 * @company www.itheima.com
 */
public interface CartService {
    /**
     * 向已有的购物列表中 添加商品
     * @param cartList 已有的购物车列表
     * @param itemId 要添加的商品的ID
     * @param num 要添加数量
     * @return
     */
    public List<Cart> addGoodsToCartList(List<Cart> cartList,Long itemId,Integer num);

    //从redis中获取购物车的列表
    public List<Cart> getCartListFromRedis(String username);
    //向redis中存储购物车的列表
    public void saveCartListToRedis(String username,List<Cart> cartList);

    public List<Cart> mergeCartList(List<Cart> cookieList,List<Cart> redisList);
}
